numpy/f2py/crackfortran.py

```diff
@@ -2481,11 +2505,30 @@ def get_parameters(vars, global_params={}):
                     # FIXME, unused l looks like potential bug
                     l = markoutercomma(v[1:-1]).split('@,@')
 
-            try:
-                params[n] = eval(v, g_params, params)
-            except Exception as msg:
-                params[n] = v
-                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(v)))
+            # Identifying array parameters
+            param_is_array = False
+            for a in vars[n]['attrspec']:
+                if a[:9] == 'dimension':
+                    param_is_array = True
+
+            if param_is_array:
+                v = v.lstrip('(/').rstrip('/)').split(',')
+                # TODO Probably not the most efficient way...
+                v_eval = []
+                for item in v:
```

Comment:
I will do this in a separate PR if that's ok.

Reply:
Sorry, I ended up doing this all here since I couldn't solve one problem without touching the other. Let me know if you see any problems with the code here